app/app/mailers/weekly_report_mailer.rb

```diff
@@ -30,41 +30,47 @@ def report_filename(report_range)
   end
 
   def weekly_report_data(current_agency, report_range)
-    CbvFlowInvitation
-      .where(client_agency_id: current_agency.id)
-      .where(created_at: report_range)
-      .includes(:cbv_flows, :cbv_applicant)
-      .map do |invitation|
-        cbv_flow = invitation.cbv_flows.find(&:complete?)
-        applicant = invitation.cbv_applicant
-
-        base_fields = {
-          invited_at: invitation.created_at,
-          transmitted_at: cbv_flow&.transmitted_at,
-          completed_at: cbv_flow&.consented_to_authorized_use_at
-        }
-
-        case current_agency.id
-        when "nyc"
-          base_fields.merge(
-            client_id_number: applicant.client_id_number,
-            case_number: applicant.case_number,
-            snap_application_date: applicant.snap_application_date,
-            email_address: invitation.email_address
-          )
-        when "ma"
-          base_fields.merge(
-            agency_id_number: applicant.agency_id_number,
-            beacon_id: applicant.beacon_id,
-            snap_application_date: applicant.snap_application_date,
-            email_address: invitation.email_address
-          )
-        when "la_ldh"
-          base_fields.merge(
-            case_number: applicant.case_number,
-          )
-        end
+    client_agency_id = current_agency.id
+
+    flows = CbvFlow.where(client_agency_id: client_agency_id)
+                   .includes(:cbv_applicant, :cbv_flow_invitation)
+
+    results = flows.map do |flow|
+      applicant = flow.cbv_applicant
+      invitation = flow.cbv_flow_invitation
+
+      next if invitation && !report_range.cover?(invitation.created_at)
+
+      base_fields = {
```

Comment:
Sorry, one more thing. Could we add

```suggestion
      base_fields = {
        started_at: flow.created_at,
```